sphinx/domains/python/__init__.py

```diff
@@ -38,6 +38,16 @@
 
 # re-export objects for backwards compatibility
 # xref https://github.com/sphinx-doc/sphinx/issues/12295
+from sphinx.domains.python._annotations import (  # NoQA: F401
+    _parse_annotation,
+    _parse_arglist,
+    _parse_type_list,
+    _pseudo_parse_arglist,
+    _TypeParameterListParser,
+    parse_reftarget,
+    type_to_xref,
```

Comment:
I'm uneasy about this -- these are mainly private names, and hence fragile. ``sphinx-immaterial`` should have resilience if the names go away, and ideally not use them at all.

Also note, I am only able to find uses of ``type_to_xref`` and ``_parse_arglist`` in ``sphinx-immaterial``.

Reply:
I don't blame you about being queasy on the more private imports.

I'm currently trying to see how far down these imports go, and if I can get a working install again, then working back from there.

Reply:
I was able to get the originally linked issue's reproducibility steps to again work with these proposed changes, so that's hopeful that a subset of the changes can definitely fully resolve this issue.

I definitely agree on the private imports being more problematic to include in, so I've also created https://github.com/jbms/sphinx-immaterial/issues/345 to potentially work this from both ends.